File: HurbAlpha/HurbAlpha/Model/Result.swift
//
//  Result.swift
//  HurbAlpha
//
//  Created by Julia Rocha on 04/09/19.
//  Copyright © 2019 Julia Rocha. All rights reserved.
//

import Foundation

// MARK: - Declaration

struct Result: Codable {
    let sku: String
    let isPackage: Bool?
    let isHotel:Bool?
    let name: String
    let stars:Int?
    let url: String
    let smallDescription, resultDescription: String?
    let gallery: [Gallery]
    let address: Address
    let tags: [String]?
    let price: ResultPrice
    let featuredItem: FeaturedItem
    let category: String
    let quantityDescriptors: QuantityDescriptors
    let id: String
    let amenities: [Amenity]
    
    // MARK: - Enum Coding Keys Declaration
    
    enum CodingKeys: String, CodingKey {
        case sku, isHotel, name, url, smallDescription, isPackage, stars
        case resultDescription
        case gallery, address, tags, price, featuredItem, category, quantityDescriptors, id, amenities
    }
}

File: app/src/main/java/br/com/flyingdutchman/challenge_alpha/data/model/Result.kt
package br.com.flyingdutchman.challenge_alpha.data.model

import br.com.flyingdutchman.challenge_alpha.data.model.*
import com.google.gson.annotations.SerializedName

data class Result(
    @SerializedName("address")
    val address: Address,
    @SerializedName("amenities")
    val amenities: List<Amenity>,
    @SerializedName("category")
    val category: String,
    @SerializedName("description")
    val description: String,
    @SerializedName("featuredItem")
    val featuredItem: FeaturedItem,
    @SerializedName("gallery")
    val gallery: List<Gallery>,
    @SerializedName("id")
    val id: String,
    @SerializedName("isPackage")
    val isPackage: Boolean,
    @SerializedName("isHotel")
    val isHotel: Boolean,
    @SerializedName("name")
    val name: String,
    @SerializedName("price")
    val price: PriceX,
    @SerializedName("quantityDescriptors")
    val quantityDescriptors: QuantityDescriptors,
    @SerializedName("sku")
    val sku: String,
    @SerializedName("smallDescription")
    val smallDescription: String,
    @SerializedName("tags")
    val tags: List<String>,
    @SerializedName("url")
    val url: String,
    @SerializedName("hu_free_cancellation")
    val freeCancelation: Boolean,
    @SerializedName("stars")
    val stars: Int
)